wikipedia_cleanup/data_processing.py

```diff
@@ -158,7 +161,90 @@ def get_data(
             mapped_filters.append(changes_and_filters[1])
     all_changes = itertools.chain.from_iterable(all_changes)
     merge_filter_stats_into(mapped_filters, filters)
-    return pd.DataFrame([change.__dict__ for change in all_changes])
+    res_df = pd.DataFrame([change.__dict__ for change in all_changes])
+    sorted_columns = sorted(list(res_df.columns))
+    return res_df[sorted_columns]
+
+
+def get_data_single(
+    file: Path,
+    data_filters: Optional[List[AbstractDataFilter]] = None,
+) -> pd.DataFrame:
+    if data_filters is None:
+        data_filters = []
+    changes_and_filters = read_and_filter_file(file, data_filters)
+    all_changes = [changes_and_filters[0]]
+    mapped_filters = [changes_and_filters[1]]
+    all_changes_iterator = itertools.chain.from_iterable(all_changes)
+    merge_filter_stats_into(mapped_filters, data_filters)
+    return pd.DataFrame([change.__dict__ for change in all_changes_iterator])
+
+
+def feature_generation(df):
```

Comment:
this is very duplicated with the feature adder filter.

Reply:
Yes this is the old code. This should probably get deleted now. @BleTib